subprojects/diagnostics/src/integTest/groovy/org/gradle/api/plugins/ProjectReportsPluginIntegrationTest.java

```diff
@@ -16,18 +16,85 @@
 package org.gradle.api.plugins;
 
 import org.gradle.integtests.fixtures.AbstractIntegrationTest;
+import org.gradle.integtests.fixtures.executer.ExecutionResult;
+import org.gradle.util.TextUtil;
 import org.junit.Test;
 
 public class ProjectReportsPluginIntegrationTest extends AbstractIntegrationTest {
     @Test
     public void generatesReportFilesToReportsDirectory() {
-        testFile("build.gradle").writelns(
-                "apply plugin: 'project-report'"
-        );
+        applyProjectReportPlugin();
         inTestDirectory().withTasks("projectReport").run();
 
         testFile("build/reports/project/dependencies.txt").assertExists();
         testFile("build/reports/project/properties.txt").assertExists();
         testFile("build/reports/project/tasks.txt").assertExists();
     }
+
+    private void applyProjectReportPlugin() {
+        testFile("build.gradle").writelns(
+            "apply plugin: 'project-report'"
+        );
+    }
+
+    @Test
```

Comment:
Ok, I did not know about the refactoring to Spock tests, although I noted the mix of JUnit and Spock. I'll move the tests to the specification class and will do the same in the future.

Thanks for clarifying that.